### PR TITLE
Fix month calculation to 30 days, and add no expiry and year as expiry options

### LINKED ISSUE #47: A month is not 30 weeks (LOL)
Simple bug in `src/common.js`

Second last line in this snippet
```JavaScript
let expirationSeconds = parseFloat(expirationStr)
  const lastChar = expirationStr[expirationStr.length - 1]
  if (lastChar === 'm') expirationSeconds *= 60
  else if (lastChar === 'h') expirationSeconds *= 3600
  else if (lastChar === 'd') expirationSeconds *= 3600 * 24
  else if (lastChar === 'w') expirationSeconds *= 3600 * 24 * 7
  else if (lastChar === 'M') expirationSeconds *= 3600 * 24 * 7 * 30
  return expirationSeconds
```

### LINKED ISSUE #48: [Feature Request] Allow no expiry and specifying expiry in years

### PR DESCRIPTION
**Pull request by rutvora**
Fixes #47, Resolves #48

### CODE REVIEW

**Comment by rutvora** [comment]
Sorry for closing and re-opening the PR (it was a mis-click)

**Comment by SharzyL** [comment]
Seems that you also need to modify the corresponding tests

**Comment by rutvora** [comment]
> Seems that you also need to modify the corresponding tests

Done

File: test/basic.spec.js [hunk 1]
```diff
@@ -127,7 +127,7 @@ test("expire", async () => {
   await testExpireParse("100m", 6000)
   await testExpireParse("100h", 360000)
   await testExpireParse("1d", 86400)
-  await testExpireParse("1M", 18144000)
+  await testExpireParse("1M", 2592000)
   await testExpireParse("100  m", 6000)
```
Comment (SharzyL) [inline test/basic.spec.js:132]: We also need tests on `Y` and empty expiration, considering changes in `parseExpiration()`

File: frontend/index.html [hunk 2]
```diff
@@ -43,7 +43,7 @@ <h2>Settings</h2>
         <div id='paste-expiration-panel' class='paste-setting-subitem-panel'>
             <input list='expiration-choices' type='text' min='60' step='1' name='paste-expiration'
                    id='paste-expiration-input' placeholder='Expiration (secs)' value='7d'>
-            <label class='small-label' for='paste-expiration-input'>Delete your paste after a period of time. <br>Units: s (seconds), m (minutes), h (hours), d (days), M (months)</label>
+            <label class='small-label' for='paste-expiration-input'>Delete your paste after a period of time (leave blank for no expiry). <br>Units: s (seconds), m (minutes), h (hours), d (days), M (months), Y (years) <br> Max 68 years</label>
```
Comment (SharzyL) [inline frontend/index.html]: `Max 68 years` could be omitted since I don't think the service can live so long and I prefer the label to be as concise as possible.

File: frontend/index.client.js [hunk 3]
```diff
@@ -83,7 +83,10 @@ window.addEventListener('DOMContentLoaded', () => {
     const pasteNotEmpty = inputType === 'edit'
       ? pasteEditArea.prop('value').length > 0
       : file !== null
-    const expirationValid = EXPIRE_REGEX.test(expiration)  // TODO: verify it
+    expirationValid = EXPIRE_REGEX.test(expiration)  // TODO: verify it
```
Comment (SharzyL) [inline frontend/index.client.js]: Use `let`, no implicit declaration
